src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,107 @@
+                break;
+            case "list":
+                str = tasklist.list();
+                break;
+            case "mark":
+                if (!details) {
+                    throw new DukeException("Please include the task index to mark");
+                } else {
+                    str = tasklist.setTaskStatus(Integer.parseInt(arr[1]), true);
+                    break;
+                }
+            case "unmark":
+                if (!details) {
+                    throw new DukeException("Please include the task index to unmark.");
+                }
+                str = tasklist.setTaskStatus(Integer.parseInt(arr[1]), false);
+                break;
+            case "delete":
+                if (!details) {
+                    throw new DukeException("Please include the task index to delete.");
+                }
+                str = tasklist.delete(Integer.parseInt(arr[1]));
+                break;
+            case "todo":
+                if (!details) {
+                    throw new DukeException("Please include the todo details.");
+                }
+                str = tasklist.addToDo(arr[1]);
+                break;
+            case "deadline":
+                if (!details) {
+                    throw new DukeException("Please include the deadline details.");
+                }
+                String[] descriptionBy = arr[1].split("/by", 2);
+                if (descriptionBy.length == 1) {
+                    throw new DukeException("Please insert deadline date after /by");
+                }
+
+                str = tasklist.addDeadline(descriptionBy[0], descriptionBy[1]);
+                break;
+            case "event":
+                if (!details) {
+                    throw new DukeException("Please include the event details.");
+                }
+                String[] descriptionOthers = arr[1].split("/from", 2);
+                if (descriptionOthers.length == 1) {
+                    throw new DukeException("Please insert the date the event takes place from, after /from ");
+                }
+                String[] fromTo = descriptionOthers[1].split("/to", 2);
+                if (fromTo.length == 1) {
+                    throw new DukeException("Please insert the date the event takes place until, after /to ");
+                }
+                str = tasklist.addEvent(descriptionOthers[0], fromTo[0], fromTo[1]);
+                break;
+            case "find":
+                if (!details) {
+                    throw new DukeException("Please include the task you would like to find.");
+                }
+                str = tasklist.find(arr[1]);
+                break;
+            default:
+                throw new DukeException("Sorry, I don't know what that means.");
+            }
+
+            // handle changes to arraylist
+            storage.writeFile(tasklist);
+        } catch (DukeException e) {
+            str = ui.showError(e);
+
```

Comment:
There might be an additional space that was accidentally entered. Not big of an issue.

```suggestion
```